portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -84,11 +85,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
      */
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String name = request.getParameter("name");
```

Comment:
I think you may forget to escape the value for all input boxes. I tested your site by putting "<script>alert("escaped?")</script>" as the comment. Now your site will always popup an alert when it gets loaded.

Reply:
added function to escape html characters before passing data to backend